src/apps/clients/notifications.clj

```diff
@@ -49,6 +51,18 @@
   (boolean (and (:notify job-info false)
                 (emailable-job-statuses (:status job-info)))))
 
+
+(defn- interapps-url
+  "Returns the externally accessible URL to the interactive app as a URL from
+   cemerick.url.
+
+   Example usage:
+       (str (interapps-url (url (config/interapps-base)) job-info))
+     Returns:
+       https://abb9730df.cyverse.run"
+  [{host :host :as base} {user-id :user_id analysis-id :uuid}]
+  (assoc base :host (str "a" (-> (str user-id analysis-id) sha256 (subs 0 8)) "." host)))
```

Comment:
Is prepending "a" to the sha256 required?

Reply:
Yup. The subdomain gets added as a host for a Kubernetes Ingress, which throws an error if the host doesn't start with a letter. I was originally prepending "app-" to the subdomain, but after talking with Sriram and Sarah I decided to shorten it even further.